streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -88,57 +115,53 @@ public ProcessorStateManager(final TaskId taskId,
                                  final Map<String, String> storeToChangelogTopic,
                                  final ChangelogReader changelogReader,
                                  final LogContext logContext) throws ProcessorStateException {
+        this.logPrefix = format("task [%s] ", taskId);
         this.log = logContext.logger(ProcessorStateManager.class);
+
         this.taskId = taskId;
+        this.sourcePartitions = sources;
         this.changelogReader = changelogReader;
-        this.logPrefix = format("task [%s] ", taskId);
+        this.storeToChangelogTopic = storeToChangelogTopic;
 
-        partitionForTopic = new HashMap<>();
-        for (final TopicPartition source : sources) {
-            partitionForTopic.put(source.topic(), source);
-        }
-        offsetLimits = new HashMap<>();
-        standbyRestoredOffsets = new HashMap<>();
-        this.isStandby = isStandby;
-        restoreCallbacks = isStandby ? new HashMap<>() : null;
-        recordConverters = isStandby ? new HashMap<>() : null;
-        this.storeToChangelogTopic = new HashMap<>(storeToChangelogTopic);
+        this.baseDir = stateDirectory.directoryForTask(taskId);
+        this.checkpointFile = new OffsetCheckpoint(new File(baseDir, CHECKPOINT_FILE_NAME));
 
-        baseDir = stateDirectory.directoryForTask(taskId);
-        checkpointFile = new OffsetCheckpoint(new File(baseDir, CHECKPOINT_FILE_NAME));
+        log.debug("Created state store manager for task {}", taskId);
+    }
 
+    public void initialize() {
```

Comment:
This is to fix https://issues.apache.org/jira/browse/KAFKA-8574: we do not try to read checkpoint file at constructor anymore, but at `initialize`, which is called by the task upon `initializeStateStores` in which we would:

1) try to grab locks
2) stateManager.initialize()
3) register each state store into the stateManager

cc @vvcephei @ableegoldman